api/layer/cors.go

```diff
@@ -48,13 +50,13 @@ func (n *layer) PutBucketCORS(ctx context.Context, p *PutCORSParams) error {
 		return err
 	}
 
-	objIDToDelete, err := n.treeService.PutBucketCORS(ctx, &p.BktInfo.CID, objID)
+	objIDToDelete, err := n.treeService.PutBucketCORS(ctx, p.BktInfo.CID, objID)
 	if err != nil {
 		return err
 	}
 
-	if objIDToDelete != nil {
-		if err = n.objectDelete(ctx, p.BktInfo, *objIDToDelete); err != nil {
+	if !objIDToDelete.Equals(oid.ID{}) {
```

Comment:
I think such an expected behavior should be reflected in `TreeService.PutBucketCORS` docs. I also suggest to not compare ID with zero value, but return some declared error instead (docs too).

Same for `DeleteBucketCORS`, `AddPart` and all other similar places.